client/tso_dispatcher.go

```diff
@@ -138,12 +138,34 @@ func (c *tsoClient) updateTSODispatcher() {
 	})
 }
 
-type deadline struct {
-	timer  <-chan time.Time
+var timerPool = sync.Pool{
+	New: func() interface{} {
+		return time.NewTimer(defaultPDTimeout)
+	},
+}
+
+// TSDeadline is used to watch the deadline of each tso request.
+type TSDeadline struct {
+	timer  *time.Timer
 	done   chan struct{}
 	cancel context.CancelFunc
 }
 
+// NewTSDeadline creates a new TSDeadline.
+func NewTSDeadline(
+	timeout time.Duration,
+	done chan struct{},
+	cancel context.CancelFunc,
+) *TSDeadline {
+	timer := timerPool.Get().(*time.Timer)
+	timer.Reset(timeout)
```

Comment:
Do we need to reset here?

Reply:
yes, timer needs to be Reset.
> Reset stops a ticker and resets its period to the specified duration. The next tick will arrive after the new period elapses
https://pkg.go.dev/time@master#Ticker.Reset